src/trapfunc.cpp

```diff
@@ -1337,6 +1337,27 @@ bool trapfunc::temple_toggle( const tripoint &p, Creature *c, item * )
                     }
                 }
             }
+            
+            // In case we're completely encirled by walls, replace random wall around the player with floor tile
```

Comment:
This comment is misleading, because it always does this, it doesn't check if the character is actually blocked in before doing so.

Reply:
Huh? This code snippet checks exactly that

```
        std::vector<tripoint> blocked_tiles;
            for( const tripoint &pnt : here.points_in_radius( c->pos(), 1 ) ) {
                if( here.passable( pnt ) ) {
                    break;
                }
                blocked_tiles.push_back( pnt );
            }
```

If there's at least one passable tile around player, then bail out and do nothing. Otherwise (i.e. there's no passable tiles around player), fill vector with positions of non-passable tiles and later replace of them with a passable one.

Reply:
If the first tile it checks is not passable it will add that to the vector, so the vector will not be empty when it breaks on a later passable tile. Also, points_in_radius should check the tile the character is on, which you don't want to do in this check.

Reply:
Resolved in [5f3fb2d](https://github.com/CleverRaven/Cataclysm-DDA/pull/60046/commits/5f3fb2d01ba5ca40a0fdd653b2a7ab682aa21b81).